cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexUpdater.java

```diff
@@ -139,9 +135,9 @@ private boolean updateIndex(final String indexName,
             return false;
         }
 
-        Future<Boolean> result = queue.submit( new Callable<Boolean>() {
+        Future<Boolean> result = queue.submitTransaction(new SQLQueueCallable<Boolean>() {
```

Comment:
Now that this is using a `SQLQueueCallable` and `submitTransaction` it is strange that it also calls begin transaction. I think we should either just use `submit` and let it handle the transaction or preferably remove the extra transaction calls and let `submitTransaction` control the transaction (I think we just need to throw where currently we set `transactionSuccess=false`). At the moment we have two sets of calls trying to do the same thing.